airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/adapters.py

```diff
@@ -372,8 +386,19 @@ def generate(self) -> Iterable[Partition]:
 
         :return: An iterable of StreamPartition objects.
         """
-        for slice_start, slice_end in self._cursor.generate_slices():
-            stream_slice = StreamSlice(partition={}, cursor_slice={"start": slice_start, "end": slice_end})
+
+        start_boundary = self._slice_boundary_fields[self._START_BOUNDARY] if self._slice_boundary_fields else "start"
+        end_boundary = self._slice_boundary_fields[self._END_BOUNDARY] if self._slice_boundary_fields else "end"
+
+        wam = list(self._cursor.generate_slices())
+        for slice_start, slice_end in wam:
```

Comment:
nit: It seems like it would be more memory efficient to directly iterate over the generated slices, is there a specific reason for saving to a list?

Reply:
correct. I had originally added this for debugging to see the entire set of slices easier, but you are correct this should be iterable